src/volunteer-form/index.tsx

```diff
@@ -367,19 +367,20 @@ class VolunteerForm extends React.Component<
         this.props.aspect === 'tutoring' || this.state.submittingTutor,
     });
 
-    const result = (await this.context.signup(
-      this.context.user
-    )) as ApiCallResult;
-    if (result.code === 201) {
-      this.setState({
-        submittedMentor:
-          this.props.aspect === 'mentoring' || this.state.submittedMentor,
-        submittedTutor:
-          this.props.aspect === 'tutoring' || this.state.submittedTutor,
-      });
+    const [error] = await to<null, ApiCallError>(
+      this.context.signup(this.context.user)
```

Comment:
You'll also wanted to destructure context (this is another one of our new ESLint rules) like so:
```typescript
const { signup, user } = this.context;
const [error] = await to<null, ApiCallError>(signup(user));
```